control-plane-operator/controllers/hostedcontrolplane/v2/kas/params.go

```diff
@@ -118,6 +118,10 @@ func NewConfigParams(hcp *hyperv1.HostedControlPlane) KubeAPIServerConfigParams
 		kasConfig.MaxMutatingRequestsInflight = mutatingReqInflight
 	}
 
+	if capabilities.IsImageRegistryCapabilityEnabled(hcp.Spec.Capabilities) {
```

Comment:
v1 kas doesn't seem to have this change. Also you made a change to v1 oapi but not v2. I assume you meant to do that here instead of the kas change?

Reply:
for oapi v2, this is defined in the config manifest directly https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/v2/assets/openshift-apiserver/config.yaml#L44

so in the code you need to set this to empty string when the capability is disabled.

Reply:
Thanks for the pointer, hopefully I covered everything this time around, PTAL

Reply:
thanks! looks good